packages/ripple-binary-codec/src/enums/rippled-definitions.ts

```diff
@@ -0,0 +1,267 @@
+import { SerializedType } from '../types/serialized-type'
+import { Buffer } from 'buffer/'
+import { BytesList, BinaryParser } from '../binary'
+
+const TYPE_WIDTH = 2
+const LEDGER_ENTRY_WIDTH = 2
+const TRANSACTION_TYPE_WIDTH = 2
+const TRANSACTION_RESULT_WIDTH = 1
+
+interface DefinitionsData {
+  TYPES: Record<string, number>
+  LEDGER_ENTRY_TYPES: Record<string, number>
+  FIELDS: (string | FieldInfo)[][]
+  TRANSACTION_RESULTS: Record<string, number>
+  TRANSACTION_TYPES: Record<string, number>
+}
+
+/*
+ * @brief: Serialize a field based on type_code and Field.nth
+ */
+function fieldHeader(type: number, nth: number): Buffer {
```

Comment:
All these other methods/classes that aren't `RippledDefinitions` should be in a separate file